crates/uv-python/src/managed.rs

```diff
@@ -669,6 +655,239 @@ impl ManagedPythonInstallation {
+    /// the symlink directory would be `/path/to/uv/python/cpython-3.10-macos-aarch64-none` and the executable path including the
+    /// symlink directory would be `/path/to/uv/python/cpython-3.10-macos-aarch64-none/bin/python3.10`.
+    ///
+    /// ## Windows
+    /// For a Python 3.10.8 installation in `C:\path\to\uv\python\cpython-3.10.8-windows-x86_64-none\python.exe`,
+    /// the junction would be `C:\path\to\uv\python\cpython-3.10-windows-x86_64-none` and the executable path including the
+    /// junction would be `C:\path\to\uv\python\cpython-3.10-windows-x86_64-none\python.exe`.
+    pub fn from_executable(
+        executable: &Path,
+        key: &PythonInstallationKey,
+        preview: PreviewMode,
+    ) -> Option<Self> {
+        let implementation = key.implementation();
+        if !matches!(
+            implementation,
+            LenientImplementationName::Known(ImplementationName::CPython)
+        ) {
+            // We don't currently support transparent upgrades for PyPy or GraalPy.
+            return None;
+        }
+        let executable_name = executable
+            .file_name()
+            .expect("Executable file name should exist");
+        let symlink_directory_name = PythonInstallationMinorVersionKey::ref_cast(key).to_string();
+        let parent = executable
+            .parent()
+            .expect("Executable should have parent directory");
+
+        // The home directory of the Python installation
+        let target_directory = if cfg!(unix) {
+            if parent
+                .components()
+                .next_back()
+                .is_some_and(|c| c.as_os_str() == "bin")
+            {
+                parent.parent()?.to_path_buf()
+            } else {
+                return None;
+            }
+        } else if cfg!(windows) {
+            parent.to_path_buf()
+        } else {
+            unimplemented!("Only Windows and Unix systems are supported.")
+        };
+        let symlink_directory = target_directory.with_file_name(symlink_directory_name);
+        // If this would create a circular link, return `None`.
+        if target_directory == symlink_directory {
+            return None;
+        }
+        // The full executable path including the symlink directory (or junction).
+        let symlink_executable = executable_path_from_base(
+            symlink_directory.as_path(),
+            &executable_name.to_string_lossy(),
+            implementation,
+        );
+        let minor_version_link = Self {
+            symlink_directory,
+            symlink_executable,
+            target_directory,
+        };
+        // If preview mode is disabled, still return a `MinorVersionSymlink` for
+        // existing symlinks, allowing continued operations without the `--preview`
+        // flag after initial symlink directory installation.
+        if preview.is_disabled() {
```

Comment:
This nesting is left over from an earlier version and I'm going to consolidate into
```
if preview.is_disabled() && !minor_version_link.symlink_exists() {
    return None;
}
```